src/Controller/Component/CsrfComponent.php

```diff
@@ -61,6 +61,19 @@ class CsrfComponent extends Component
         'field' => '_csrfToken',
     ];
 
+    /**
+     * Warn if CsrfComponent is used together with CsrfProtectionMiddleware
+     *
+     * @param array $config The config data.
+     * @return void
+     */
+    public function initialize(array $config)
+    {
+        if ($this->getController()->getRequest()->getParam('_csrfToken') !== false) {
```

Comment:
Won't this emit errors if the value is null?

Reply:
Which one?
`getParam('_csrfToken')` returns `false` if it is not set.
`getController()` and `getRequest()` are always set.

Reply:
Ok. I thought getParam() could return null if the parameter was not set. I'm probably conflating 3.x and 4 though.

Reply:
`ServerRequest::getParam()` returning `false` (since the default value for 2nd param is `false`) when key does not exist is a wacky quirk of 3.x which is fixed in 4.x.